library/src/main/java/com/novoda/downloadmanager/lib/DownloadReceiver.java

```diff
@@ -102,31 +106,33 @@ public void run() {
     }
 
     private void handleNotificationBroadcast(Context context, Intent intent) {
-        String action = intent.getAction();
         long[] ids = intent.getLongArrayExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_IDS);
         int[] statuses = intent.getIntArrayExtra(DownloadManager.EXTRA_NOTIFICATION_CLICK_DOWNLOAD_STATUSES);
         long batchId = getBatchId(intent);
+
+        String action = intent.getAction();
         switch (action) {
-            case NotificationDisplayer.ACTION_LIST:
+            case NotificationDisplayer.ACTION_DOWNLOAD_CANCELLED_CLICK:
+            case NotificationDisplayer.ACTION_DOWNLOAD_FAILED_CLICK:
+            case NotificationDisplayer.ACTION_DOWNLOAD_SUCCESS_CLICK:
                 sendNotificationClickedIntent(context, ids, statuses);
-                if (DownloadStatus.isFailure(statuses[0])) {
```

Comment:
this didn't work cause the status was translated - now there's no need to check for the status here, we can just check the action :boom: